src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1741,6 +1741,24 @@ def _gen_guid():
     return uuid.uuid4()
 
 
+def _random_password(length):
+    import random
+    import string
+    random_source = string.ascii_letters + string.digits + string.punctuation
+    password = random.choice(string.ascii_lowercase)
+    password += random.choice(string.ascii_uppercase)
+    password += random.choice(string.digits)
+    password += random.choice(string.punctuation)
+
+    for i in range(length - 4):
+        password += random.choice(random_source)
+
+    password_list = list(password)
+    random.SystemRandom().shuffle(password_list)
+    password = ''.join(password_list)
+    return password
```

Comment:
If the code is borrowed from somewhere else, it is a good practice to put the source link in the comment: https://pynative.com/python-generate-random-string/